src/frontend/src/instance.rs

```diff
@@ -370,12 +391,25 @@ impl Instance {
     pub fn plugins(&self) -> Arc<Plugins> {
         self.plugins.clone()
     }
+
+    pub async fn shutdown(&self) -> Result<()> {
+        futures::future::try_join_all(self.servers.values().map(|server| server.0.shutdown()))
+            .await
+            .context(error::ShutdownServerSnafu)?;
+
+        Ok(())
+    }
 }
 
 #[async_trait]
 impl FrontendInstance for Instance {
     async fn start(&mut self) -> Result<()> {
         // TODO(hl): Frontend init should move to here
+
+        futures::future::try_join_all(self.servers.values().map(start_server))
+            .await
+            .context(error::StartServerSnafu)?;
+
         Ok(())
```

Comment:
```suggestion
        futures::future::try_join_all(self.servers.values().map(start_server))
            .await
            .context(error::StartServerSnafu)
```

Reply:
> Suggested change
>         futures::future::try_join_all(self.servers.values().map(start_server))
>             .await
>             .context(error::StartServerSnafu)?;
>         Ok(())
>         futures::future::try_join_all(self.servers.values().map(start_server))
>             .await
>             .context(error::StartServerSnafu)

Removing the `Ok(())` will lead to a type error. or mapping `Vec<()>` to `()`  ?
```rust
        futures::future::try_join_all(self.servers.values().map(start_server))
            .await
            .context(error::StartServerSnafu)
            .map(|_| ())
```

Reply:
Yep. It's my mistake. Ignore it.